contracts/governance/src/ZeroExVotes.sol

```diff
@@ -168,10 +195,20 @@ contract ZeroExVotes is IZeroExVotes {
         oldWeight = oldCkpt.votes;
         newWeight = op(oldWeight, delta);
 
+        uint256 oldQuadraticWeight = oldCkpt.quadraticVotes;
+        uint256 newQuadraticWeight = op(oldQuadraticWeight, Math.sqrt(delta));
```

Comment:
This doesn't give the correct value. Imagine Alice has 4 tokens, Bob has 0, and delta is 1. `sqrt(delta)` is 1, so at the end Bob would have 1 quadratic vote (correct) but Alice would have 3 quadratic votes, when she should have `sqrt(3) ~= 1.732`.

Reply:
In your example `delta` is 4 not 1 though.

Reply:
Okay so you mean in the case of transferring 1 token. Alices quadratic votes initial balance will be 2 not 3 ending up with 1 quadratic vote after the transfer which is correct minus the rounding down error.

Reply:
There will be rounding errors in that though I agree.

Reply:
This isn't just a rounding error issue. Let me be a little more explicit:

---

Starting conditions:
| Named quantity | Linear | As decimal (basis 10^18)| Quadratic | As decimal (basis 10^9) |
| :-- | :-- | :-- | :-- | :-- |
| Alice balance | 4ether  | 4.0 | 2gwei | 2.0 |
| Bob balance | 0 | 0.0 | 0 | 0.0 |
| delta | 1ether | 1.0 | 1gwei | 1.0 |

---

End conditions: 

| Named quantity | Linear | As decimal (basis 10^18) | Quadratic | As decimal (basis 10^9) |
| :-- | :-- | :-- | :-- | :-- |
| Alice balance | 3ether | 3.0 | 1gwei | 1.0 |
| Bob balance | 1ether | 1.0 | 1gwei | 1.0 |

Notice how Alice's quadratic balance is not equal to the sqrt of her linear balance. This is because the calculation on the referenced line is incorrect.

---

Correct end condition: 

| Named quantity | Linear | As decimal (basis 10^18) | Quadratic | As decimal (basis 10^9) |
| :-- | :-- | :-- | :-- | :-- |
| Alice balance | 3ether | 3.0 | 1732050807 | ~1.732 |
| Bob balance | 1ether | 1.0 | 1gwei | 1.0 |

Reply:
I see now thanks

Reply:
Just circling back to the high level quadratic voting logic and specifically calculating voting power of delegates, e.g. if person A holding X tokens and person B holding Y tokens have delegated their tokens to person C, then person C's voting power in a QV vote when calculated as `sqrt(X)+sqrt(Y)` rather than `sqrt(X+Y)` can potentially be gamed if the delegates split their balances between multiple accounts right?

Reply:
Yes. This property is inherent to quadratic voting. Avoiding this would require adoption of advanced anti-collusion techniques, which we intend to adopt but are beyond the complexity acceptable for V1